src/jit/importer.cpp

```diff
@@ -13907,15 +13907,23 @@ bool Compiler::impReturnInstruction(BasicBlock *block, int prefixFlags, OPCODE &
 
         op2 = impAssignStructPtr(retBuffAddr, op2, retClsHnd, (unsigned)CHECK_SPILL_ALL);
         impAppendTree(op2, (unsigned)CHECK_SPILL_NONE, impCurStmtOffs);
-        if (compIsProfilerHookNeeded())
+
+        // There are cases where the address of the implicit RetBuf should be returned in RAX.  
+#if !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+        // 1. In case of Windows AMD64 the profiler hook requires to have the RAX contain the implicit RetBuf.  
+        //    In such case the return value of the function is changed to BYREF.  
+        //    If profiler hook is not needed the return type of the function is TYP_VOID.  
+        if (!compIsProfilerHookNeeded())
         {
```

Comment:
Please don't change the sense of this if condition. Checking the negation of a condition and then having an else clause is confusing. Also, in the else clause you have commented code that is not only unix specific with a target-specific comment. I think it would be better to fully separate the unix case.